packages/next/next-server/server/config.ts

```diff
@@ -10,7 +10,9 @@ import * as Log from '../../build/output/log'
 const targets = ['server', 'serverless', 'experimental-serverless-trace']
 const reactModes = ['legacy', 'blocking', 'concurrent']
 
-const defaultConfig: { [key: string]: any } = {
+type Config = { [key: string]: any }
```

Comment:
This type seems too loose to be of use.

Reply:
True, the intention of this PR is to have a type for the Config, the same one was being reused in several places on the file. next step will be to replace it for a more declarative type for each one of the configs options.

Reply:
Let me do a few changes to this PR, including the proper types

Reply:
Happy to land if this object is a strict full type.

Reply:
My original plan was to introduce a strict interface like, what do you think?

```
interface Config {
  env: any[];
  webpack: any | null;
  webpackDevMiddleware: any | null;
  distDir: string;
  assetPrefix: string;
  configOrigin: string;
  useFileSystemPublicRoutes: boolean;
  generateBuildId: () => null;
  generateEtags: boolean;
  pageExtensions: string[];
  target: string;
  poweredByHeader: boolean;
  compress: boolean;
  devIndicators: {
    buildActivity: boolean;
    autoPrerender: boolean;
  };
  onDemandEntries: {
    maxInactiveAge: number;
    pagesBufferLength: number;
  };
  amp: {
    canonicalBase: string;
  };
  exportTrailingSlash: boolean;
  sassOptions: object;
  experimental: {
    cpus: number;
    granularChunks: boolean;
    modern: boolean;
    plugins: boolean;
    profiling: boolean;
    sprFlushToDisk: boolean;
    reactMode: string;
    workerThreads: boolean;
    basePath: string;
    pageEnv: boolean;
    productionBrowserSourceMaps: boolean;
    optionalCatchAll: boolean;
  };
  future: {
    excludeDefaultMomentLocales: boolean;
  };
  serverRuntimeConfig: object;
  publicRuntimeConfig: object;
  reactStrictMode: boolean;
}
```

Reply:
That looks pretty good to me. Let's not type `experimental` strictly tho.